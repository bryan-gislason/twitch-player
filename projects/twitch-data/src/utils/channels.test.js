import { extractChannelsForCategory, extractChannelsForPlatformType } from ".";

describe("extractChannelsForCategory", () => {
  const { data } = require("../../fixtures/categoryChannels.json");
  const similarData = {
    ...data,
    category: {
      ...data.category,
      channels: {
        ...data.category.channels,
        edges: data.category.channels.edges.slice(),
      },
    },
  };
  const dissimilarData = {
    ...data,
    category: {
      ...data.category,
      channels: {
        ...data.category.channels,
        edges: data.category.channels.edges.slice(1),
      },
    },
  };

  it("extracts the channel nodes from a response", () => {
    expect(extractChannelsForCategory(data)).toMatchSnapshot();
  });

  it("returns the same result for the exact same argument", () => {
    expect(extractChannelsForCategory(data)).toBe(
      extractChannelsForCategory(data)
    );
  });

  it("returns the same result for similar argument", () => {
    expect(extractChannelsForCategory(data)).toBe(
      extractChannelsForCategory(similarData)
    );
  });

  it("returns a different result for dissimilar argument", () => {
    expect(extractChannelsForCategory(data)).not.toBe(
      extractChannelsForCategory(dissimilarData)
    );
  });
});

describe("extractChannelsForPlatformType", () => {
  const { data } = require("../../fixtures/topChannels.json");
  const similarData = {
    ...data,
    channels: {
      ...data.channels,
      edges: data.channels.edges.slice(),
    },
  };
  const dissimilarData = {
    ...data,
    channels: {
      ...data.channels,
      edges: data.channels.edges.slice(1),
    },
  };

  it("extracts the channel nodes from a response", () => {
    expect(extractChannelsForPlatformType("all")(data)).toMatchSnapshot();
  });

  it("returns the same result for the exact same argument", () => {
    expect(extractChannelsForPlatformType("all")(data)).toBe(
      extractChannelsForPlatformType("all")(data)
    );
  });

  it("returns the same result for similar argument", () => {
    expect(extractChannelsForPlatformType("all")(data)).toBe(
      extractChannelsForPlatformType("all")(similarData)
    );
  });

  it("returns a different result for dissimilar argument", () => {
    expect(extractChannelsForPlatformType("all")(data)).not.toBe(
      extractChannelsForPlatformType("all")(dissimilarData)
    );
  });

  it("returns a different result for different platform", () => {
    expect(extractChannelsForPlatformType("ps4")(data)).not.toBe(
      extractChannelsForPlatformType("xbox")(data)
    );
  });
});
