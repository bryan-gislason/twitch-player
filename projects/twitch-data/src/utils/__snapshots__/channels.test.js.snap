// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractChannelsForCategory extracts the channel nodes from a response 1`] = `
Array [
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Pop-Up w/ Cloak, Nate, & Tennp0",
      },
      "displayName": "Tfue",
      "id": "60056333",
      "key": "60056333",
      "login": "tfue",
      "profileImageURL": "profiles/2470b5c6-a737-4ba6-8987-c28e0ca839e1-profile_image-150x150.jpg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32503708720",
    "key": "32503708720",
    "previewImageURL": "channels/live_user_tfue-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 66901,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "what do ya think [ please use code \\"72hrs\\" in the item shop ]",
      },
      "displayName": "72hrs",
      "id": "101400190",
      "key": "101400190",
      "login": "72hrs",
      "profileImageURL": "profiles/dc63c4ad-1edf-46d3-8a80-0eba7d68cbfc-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32502670912",
    "key": "32502670912",
    "previewImageURL": "channels/live_user_72hrs-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 8379,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "high kill popup squads",
      },
      "displayName": "cloakzy",
      "id": "81687332",
      "key": "81687332",
      "login": "cloakzy",
      "profileImageURL": "profiles/320226c6-f422-4baf-8ed2-1be7eb3757e6-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32503267648",
    "key": "32503267648",
    "previewImageURL": "channels/live_user_cloakzy-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 8098,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Aim so Good, Kids think I Cheat (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧ ~ New Snugglez Emote out now!!!!  ~ Face Reveal @4200 Subs",
      },
      "displayName": "TUFFYtv",
      "id": "272917247",
      "key": "272917247",
      "login": "tuffytv",
      "profileImageURL": "profiles/efbcb6ad-8f4c-45ec-b6a2-00568395565d-profile_image-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32505827344",
    "key": "32505827344",
    "previewImageURL": "channels/live_user_tuffytv-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "ff56eeeb-99ed-4a60-93fc-0b3f05d8661e",
        "isLanguageTag": false,
        "key": "ff56eeeb-99ed-4a60-93fc-0b3f05d8661e",
        "localizedName": "Multiplayer",
        "tagName": "Multiplayer",
      },
      Object {
        "__typename": "Tag",
        "id": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "isLanguageTag": false,
        "key": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "localizedName": "Competitive",
        "tagName": "Competitive",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 4358,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "LIQUID SQUAD POPUP | !social",
      },
      "displayName": "POACH",
      "id": "45143025",
      "key": "45143025",
      "login": "poach",
      "profileImageURL": "profiles/6b978024-9047-4555-b68f-ae2dfdcc041b-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32505051616",
    "key": "32505051616",
    "previewImageURL": "channels/live_user_poach-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 3668,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Will shave eyebrows at 10k - Fill Kid Friday (Drunk Edition)",
      },
      "displayName": "PaymoneyWubby",
      "id": "38251312",
      "key": "38251312",
      "login": "paymoneywubby",
      "profileImageURL": "profiles/abf443ac-20dc-490d-9111-b8aaf48139c6-profile_image-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32506274096",
    "key": "32506274096",
    "previewImageURL": "channels/live_user_paymoneywubby-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 2767,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Pop up cup W/ Nick Eh 30, Hogman & Mackwood | 86,000+ Kills, 2,300+ Wins",
      },
      "displayName": "Replays",
      "id": "146790215",
      "key": "146790215",
      "login": "replays",
      "profileImageURL": "profiles/5169e81b-efdc-46aa-8d22-946f674f0844-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32502094704",
    "key": "32502094704",
    "previewImageURL": "channels/live_user_replays-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 2447,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Code Nate | Squad popup cup",
      },
      "displayName": "NateHill",
      "id": "181224914",
      "key": "181224914",
      "login": "natehill",
      "profileImageURL": "profiles/ecb96deac93fa0f7-profile_image-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32502100688",
    "key": "32502100688",
    "previewImageURL": "channels/live_user_natehill-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 2171,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "pop up best girl squad EVA (I still think you have a nice butt) MaddyNF/Chica/Kitzi",
      },
      "displayName": "KittyPlays",
      "id": "39627315",
      "key": "39627315",
      "login": "kittyplays",
      "profileImageURL": "profiles/eddda73f-f373-4920-86b4-64016ff1cbb3-profile_image-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32503164512",
    "key": "32503164512",
    "previewImageURL": "channels/live_user_kittyplays-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 1545,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "🔥Lets go, es geht wieder los! Fifa, Marbles, FN & LoL! Packs, SBCs, Fun-League & Giveaways! Pog! Icon-SBCs!",
      },
      "displayName": "LPMassive",
      "id": "40621899",
      "key": "40621899",
      "login": "lpmassive",
      "profileImageURL": "profiles/dbf36ed7-00f4-4f85-a859-c46134e6e53a-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 38506759,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 153734,
    },
    "id": "32497722080",
    "key": "32497722080",
    "previewImageURL": "channels/live_user_lpmassive-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "9166ad14-41f1-4b04-a3b8-c8eb838c6be6",
        "isLanguageTag": true,
        "key": "9166ad14-41f1-4b04-a3b8-c8eb838c6be6",
        "localizedName": "German",
        "tagName": "auto___lang_de",
      },
    ],
    "viewersCount": 1508,
  },
]
`;

exports[`extractChannelsForPlatformType extracts the channel nodes from a response 1`] = `
Array [
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Aimbot-Up Cup Duos w/ Chokzy",
      },
      "displayName": "Tfue",
      "id": "60056333",
      "key": "60056333",
      "login": "tfue",
      "profileImageURL": "profiles/2470b5c6-a737-4ba6-8987-c28e0ca839e1-profile_image-150x150.jpg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 35800802,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 240727,
    },
    "id": "31876146512",
    "key": "31876146512",
    "previewImageURL": "channels/live_user_tfue-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 56834,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "We Gaming <3 ",
      },
      "displayName": "Ninja",
      "id": "19571641",
      "key": "19571641",
      "login": "ninja",
      "profileImageURL": "profiles/cef31105-8a6e-4211-a74b-2f0bbd9791fb-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 35800802,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 240727,
    },
    "id": "31879434944",
    "key": "31879434944",
    "previewImageURL": "channels/live_user_ninja-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 25491,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Let's get anotha one  [ @summit1g ]",
      },
      "displayName": "summit1g",
      "id": "26490481",
      "key": "26490481",
      "login": "summit1g",
      "profileImageURL": "profiles/200cea12142f2384-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Sea%20of%20Thieves-376x500.jpg",
      "followersCount": 945782,
      "id": "490377",
      "key": "490377",
      "name": "Sea of Thieves",
      "viewersCount": 27869,
    },
    "id": "31880087200",
    "key": "31880087200",
    "previewImageURL": "channels/live_user_summit1g-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 24570,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "sweaty pop up cup with lupo",
      },
      "displayName": "TimTheTatman",
      "id": "36769016",
      "key": "36769016",
      "login": "timthetatman",
      "profileImageURL": "profiles/timthetatman-profile_image-4cb867e7d0af1448-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 35800802,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 240727,
    },
    "id": "31878417536",
    "key": "31878417536",
    "previewImageURL": "channels/live_user_timthetatman-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 21579,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "learning.. :( |  @shroud on socials",
      },
      "displayName": "shroud",
      "id": "37402112",
      "key": "37402112",
      "login": "shroud",
      "profileImageURL": "profiles/7ed5e0c6-0191-4eef-8328-4af6e4ea5318-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Escape%20From%20Tarkov-376x500.jpg",
      "followersCount": 985626,
      "id": "491931",
      "key": "491931",
      "name": "Escape From Tarkov",
      "viewersCount": 23676,
    },
    "id": "31878723504",
    "key": "31878723504",
    "previewImageURL": "channels/live_user_shroud-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 18166,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "pop up cup | nickmercs.sliver.tv | !giveaway | @nickmercs on socials | @100thieves",
      },
      "displayName": "NICKMERCS",
      "id": "15564828",
      "key": "15564828",
      "login": "nickmercs",
      "profileImageURL": "profiles/1aba0e3b-069c-486f-af71-5371308eeee1-profile_image-150x150.jpg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 35800802,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 240727,
    },
    "id": "31876784656",
    "key": "31876784656",
    "previewImageURL": "channels/live_user_nickmercs-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 15221,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "super gamer boy",
      },
      "displayName": "C9Sneaky",
      "id": "24538518",
      "key": "24538518",
      "login": "c9sneaky",
      "profileImageURL": "profiles/c9sneaky-profile_image-03d2c6cbb19c9e5d-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/League%20of%20Legends-376x500.jpg",
      "followersCount": 16964608,
      "id": "21779",
      "key": "21779",
      "name": "League of Legends",
      "viewersCount": 79041,
    },
    "id": "31877317168",
    "key": "31877317168",
    "previewImageURL": "channels/live_user_c9sneaky-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 14656,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "hide n seek, Drunk Jenga, and ??????",
      },
      "displayName": "FEDMYSTER",
      "id": "39040630",
      "key": "39040630",
      "login": "fedmyster",
      "profileImageURL": "profiles/77a6745c6141b9d5-profile_image-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Just%20Chatting-376x500.jpg",
      "followersCount": 957942,
      "id": "509658",
      "key": "509658",
      "name": "Just Chatting",
      "viewersCount": 55737,
    },
    "id": "31880614880",
    "key": "31880614880",
    "previewImageURL": "channels/live_user_fedmyster-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "7a3d677a-1e09-4391-b520-1fccc0e7d6ca",
        "isLanguageTag": false,
        "key": "7a3d677a-1e09-4391-b520-1fccc0e7d6ca",
        "localizedName": "Live Vlogging",
        "tagName": "Live Vlogging",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 12447,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "RANK1 GAM3R xd",
      },
      "displayName": "xQcOW",
      "id": "71092938",
      "key": "71092938",
      "login": "xqcow",
      "profileImageURL": "profiles/xqcow-profile_image-9298dca608632101-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Dota%202-376x500.jpg",
      "followersCount": 7237030,
      "id": "29595",
      "key": "29595",
      "name": "Dota 2",
      "viewersCount": 29690,
    },
    "id": "31873200672",
    "key": "31873200672",
    "previewImageURL": "channels/live_user_xqcow-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 12426,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": ":) ",
      },
      "displayName": "AdmiralBahroo",
      "id": "40972890",
      "key": "40972890",
      "login": "admiralbahroo",
      "profileImageURL": "profiles/2579b876-ba69-4ac8-8daa-035a8eea012c-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Atlas-376x500.jpg",
      "followersCount": 77431,
      "id": "5945",
      "key": "5945",
      "name": "Atlas",
      "viewersCount": 56047,
    },
    "id": "31869961424",
    "key": "31869961424",
    "previewImageURL": "channels/live_user_admiralbahroo-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 11051,
  },
]
`;
